core-tests/shared/src/test/scala/zio/CancelableFutureSpec.scala

```diff
@@ -43,15 +43,15 @@ object CancelableFutureSpec extends ZIOBaseSpec {
         val result = roundtrip.orDie.forever
 
         assertM(Live.live(result.timeout(1.seconds)))(isNone)
-      } @@ zioTag(supervision, regression),
+      } @@ zioTag(supervision, regression) @@ ignore,
       testM("unsafeRunToFuture interruptibility") {
         for {
           runtime <- ZIO.runtime[Any]
           f        = runtime.unsafeRunToFuture(UIO.never)
           _       <- UIO(f.cancel())
```

Comment:
It looks like we are hanging on attempting to interrupt, I think because our thread is waiting for the future to complete so it can return from the interruption rather than completing execution of the future.

Reply:
We could make sure that `unsafeRunToFuture` implementation does not block, but rather, uses Scala `Promise`.

Reply:
It actually does use `Promise` already, but apparently the current implementation is not sufficient.

Reply:
The issue was that the cancelation action returned by `unsafeRunAsyncCancelable` blocks for the cancelation to complete, which we can't afford to do when we only have a single thread. Added a new variant that takes a callback we can use to complete a Scala `Promise` and implemented the blocking version in terms of it.